### ISSUE TITLE
Updating options to different value, won't update selected option's label

### ISSUE THREAD

**Issue by outerlook** [JedWatson/react-select#3878]
Hi! My problem is this:

There is a form in which my users create prices and a name for this price.
Eg. Weekday - $50, Weekend - $60
![image](https://user-images.githubusercontent.com/12937160/70405402-cd5bb500-1a1b-11ea-9e1c-a1449698b85c.png)

Later, but in same view there is a select in which users select a price to later apply this price on days.
![image](https://user-images.githubusercontent.com/12937160/70405423-ea908380-1a1b-11ea-83c9-d7e4b1fe523c.png)

However if I update the weekend price to $55, it will get options to be updated on select component, but my label won't update at all... 
[Edit: the problem is on the label that is currently on the input]
![image](https://user-images.githubusercontent.com/12937160/70405478-1ca1e580-1a1c-11ea-80e9-01af1128b308.png)

Is it expected to work like that?

**Comment by jgarplind** [comment]
This got me curious so I tried recreating the issue:

https://codesandbox.io/s/sharp-blackburn-tyl9g

Seems to work fine in a minimal reproduction, so probably the issue is related to how you read/write from/to the form variable rather than React-Select.

**Comment by outerlook** [comment]
![image](https://user-images.githubusercontent.com/12937160/70987701-2d4aff00-209f-11ea-8390-73123395c4e6.png)

Sorry, but I think you reproduced correctly the same behaviour... the label on picking options it's fine. But the currently selected label that displays on input won't change.

Thanks a lot for this repro!

**Comment by alexristich** [comment]
@outerlook Looks like you'll need to add an effect to get this to work; `value` needs to be controlled outside the select component, as changes to the `options` array that gets passed in doesn't trigger a re-render of the selected value. I think this is for a very good reason: if the `value` of the selected option changes, `react-select` can't reliably know what new value to map the old one to.

Building off of the code that @jgarplind put together, I added an effect to update the selected option when weekendPrice or weekdayPrice are updated. I also changed the `value` to use a string rather than the price (to ensure we have a unique and unchanging key to use).

Give it a try here: https://codesandbox.io/s/keen-sun-qp9rb
